mobile/src/hooks/useAuthRedirect.ts

```diff
@@ -0,0 +1,17 @@
+import { useQuery } from "@tanstack/react-query";
+import { ErrorResponse } from "../types/error.type";
+import { clearAuthSession, getAuthSession } from "../utils/storage";
+
+export default function useAuthRedirect({ navigation }) {
+    useQuery({
+    queryKey: ["session"],
+    queryFn: getAuthSession,
+    //onsuccess: stay on page
+    onError: (err: ErrorResponse) => {
```

Comment:
We should check if the error code is 401. Otherwise, we will be clearing the auth session when the request fails for any reason (which we might not want). If the error code is not 401, we should retry the request until success or a 401 error code